onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selectors.cc

```diff
@@ -352,6 +352,30 @@ bool InstanceNormalizationNodeGroupSelector::Check(const GraphViewer& graph_view
          (dt_bias == ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_INT32);
 }
 
+bool BatchNormalizationNodeGroupSelector::Check(const GraphViewer& graph_viewer,
+                                                   const Node& node,
+                                                   const std::vector<const Node*>& dq_nodes,
+                                                   const std::vector<const Node*>& q_nodes) const {
+  if (!CheckQDQNodes(graph_viewer, node, dq_nodes, q_nodes)) {
+    return false;
+  }
+
+  int32_t dt_input = dq_nodes[0]->InputDefs()[0]->TypeAsProto()->tensor_type().elem_type();
+  int32_t dt_scale = dq_nodes[1]->InputDefs()[0]->TypeAsProto()->tensor_type().elem_type();
+  int32_t dt_output = q_nodes[0]->OutputDefs()[0]->TypeAsProto()->tensor_type().elem_type();
+  if (dt_input != dt_output) {
+    return false;
+  }
```

Comment:
Should there also be a check for the bias input's type, or is it unnecessary?

Reply:
Right now, it's int8 or uint8 from Ort quantization script, actually, it supports int32 for better accuracy like what we did for Conv. Don't want to make it too tight.

---
In reply to: [1149659300](https://github.com/microsoft/onnxruntime/pull/15222#discussion_r1149659300) [](ancestors = 1149659300)